lib/repl.js

```diff
@@ -283,17 +283,23 @@ function REPLServer(prompt,
   self._domain.on('error', function debugDomainError(e) {
     debug('domain error');
     const top = replMap.get(self);
+
     internalUtil.decorateErrorStack(e);
+    const isError = internalUtil.isError(e);
```

Comment:
@TimothyGu Not sure that's the behaviour I'd want - I'd like people to (optimally) be able to override `Symbol.toStringTag` (or even the name) and take the "Error like" path if they really need to. 

I don't feel strongly about this though. If you do - let me know and I'll change it.